sonarts-core/docs/rules/no-return-promise-inside-try-block.md

```diff
@@ -0,0 +1,11 @@
+# no-return-promise-inside-try-block
+
+This rule helps not to forget await in try block with return statement.
+
+## Configuration
+
+Custom Promise libraries can be provided
```

Comment:
look at the implementation of `no-invalid-await`. We rely on `then` property to defining if something is promise. 
It looks to be pretty reliable and this way rule will not require any config
```
function hasThenMethod(type: ts.Type) {
  const thenProperty = type.getProperty("then");
  return Boolean(thenProperty && thenProperty.flags & ts.SymbolFlags.Method);
}
```